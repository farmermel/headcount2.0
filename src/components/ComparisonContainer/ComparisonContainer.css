
@media screen and (max-width: 720px) {
  #compare-container {
    flex-wrap: nowrap;
  }
}