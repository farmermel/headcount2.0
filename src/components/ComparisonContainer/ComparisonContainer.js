import React from 'react';
import Card from '../Card/Card';
import ComparisonCard from '../ComparisonCard/ComparisonCard';
import PropTypes from 'prop-types';
import './ComparisonContainer.css';

const ComparisonContainer = ({ districts, compare, comparativeAnalysis }) => {
  const selectedDistricts = Object.keys(districts).map(district => {
    districts[district].selected = true;
    return (
      <Card
        {...districts[district]}
        key={districts[district].location}
        compare={compare}
      />
    );
  });

  return (
    <div className="card-container" id="compare-container">
      {selectedDistricts[0]}
      <ComparisonCard
        comparativeAnalysis={comparativeAnalysis}
        districts={districts}
      />
      {selectedDistricts[1]}
    </div>
  );
};

ComparisonContainer.propTypes = {
  districts: PropTypes.object,
  compare: PropTypes.func,
  comparativeAnalysis: PropTypes.objectOf(PropTypes.number)
};

export default ComparisonContainer;
