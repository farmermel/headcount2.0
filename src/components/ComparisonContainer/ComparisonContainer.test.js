import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonContainer from './ComparisonContainer';
import { shallow } from 'enzyme';

describe('ComparisonContainer', () => {
  let wrapper;
  let compare = jest.fn();
  const mockDistricts = [{cow: 'moo'}];
  const mockCompAnalysis = { cow: {otherCow: 'mooooo'}};

  it('should match the snapshot', () => {
    wrapper = shallow(<ComparisonContainer districts={mockDistricts}
                                           compare={compare}
                                           comparativeAnalysis={mockCompAnalysis}/>)
    expect(wrapper).toMatchSnapshot();
  })
})