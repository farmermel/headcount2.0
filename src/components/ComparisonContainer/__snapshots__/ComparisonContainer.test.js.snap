// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    comparativeAnalysis={
        Object {
            "cow": 9,
            "duck": 1,
          }
    }
    compare={[Function]}
    districts={
        Object {
            "object1": Object {
              "cow": "moo",
              "selected": true,
            },
            "object2": Object {
              "duck": "quack",
              "selected": true,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          compare={[Function]}
          cow="moo"
          selected={true}
/>,
        <ComparisonCard
          comparativeAnalysis={
                    Object {
                              "cow": 9,
                              "duck": 1,
                            }
          }
          districts={
                    Object {
                              "object1": Object {
                                "cow": "moo",
                                "selected": true,
                              },
                              "object2": Object {
                                "duck": "quack",
                                "selected": true,
                              },
                            }
          }
/>,
        <Card
          compare={[Function]}
          duck="quack"
          selected={true}
/>,
      ],
      "className": "card-container",
      "id": "compare-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compare": [Function],
          "cow": "moo",
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparativeAnalysis": Object {
            "cow": 9,
            "duck": 1,
          },
          "districts": Object {
            "object1": Object {
              "cow": "moo",
              "selected": true,
            },
            "object2": Object {
              "duck": "quack",
              "selected": true,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compare": [Function],
          "duck": "quack",
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            compare={[Function]}
            cow="moo"
            selected={true}
/>,
          <ComparisonCard
            comparativeAnalysis={
                        Object {
                                    "cow": 9,
                                    "duck": 1,
                                  }
            }
            districts={
                        Object {
                                    "object1": Object {
                                      "cow": "moo",
                                      "selected": true,
                                    },
                                    "object2": Object {
                                      "duck": "quack",
                                      "selected": true,
                                    },
                                  }
            }
/>,
          <Card
            compare={[Function]}
            duck="quack"
            selected={true}
/>,
        ],
        "className": "card-container",
        "id": "compare-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compare": [Function],
            "cow": "moo",
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparativeAnalysis": Object {
              "cow": 9,
              "duck": 1,
            },
            "districts": Object {
              "object1": Object {
                "cow": "moo",
                "selected": true,
              },
              "object2": Object {
                "duck": "quack",
                "selected": true,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compare": [Function],
            "duck": "quack",
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
